pkg/networkpath/traceroute/tcp/tcpv4.go

```diff
@@ -86,6 +82,15 @@ func (t *TCPv4) TracerouteSequential() (*Results, error) {
 		return nil, fmt.Errorf("failed to get raw ICMP listener: %w", err)
 	}
 
+	// Create a TCP listener with port 0 to get a random port from the OS
+	// and reserve it for the duration of the traceroute
+	port, tcpListener, err := reserveLocalPort()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create TCP listener: %w", err)
+	}
+	defer tcpListener.Close()
```

Comment:
Wouldn't this free the port?

Reply:
Yes, and this is what we want here. Each time we run traceroute we're going to do it from a different ephemeral port and as soon as we return from `TracerouteSequential` we know we're done with the socket.

Reply:
Would `ListenPacket` not also give you an assigned source port that you can use?

Reply:
It doesn't. ListenPacket gives us a raw socket so we aren't bound to a specific port and see all the TCP traffic bound for the host. We can't just use `net.Listen` because with that level of abstraction the expected `SYN/ACK` is abstracted away